.github/workflows/update-documentation.yml

```diff
@@ -0,0 +1,137 @@
+              continue
+            fi
+            echo "$chart: Changes detected"
+            NUM_CHANGED_CHARTS=$((NUM_CHANGED_CHARTS + 1))
+            echo "CHANGED_CHART=\"$chart\"" >> "$GITHUB_ENV"
+          done
+          if [ "$NUM_CHANGED_CHARTS" -eq 0 ]; then
+            echo "No charts have changed"
+            exit 1
+          fi
+          if [ "$NUM_CHANGED_CHARTS" -gt 1 ]; then
+            echo "Multiple charts have changed, only update one Chart per PR"
+            exit 1
+          fi
+          if git diff --name-only HEAD~1 | grep -q "$CHANGED_CHART/values.yaml"; then
+            echo "$CHANGED_CHART: No values.yaml changes detected"
+          fi
+          echo "$CHANGED_CHART: values.yaml changes detected"
+          echo "VALUES_CHANGED=true" >> "$GITHUB_ENV"
+
+  update-documentation:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    needs: gather-information
+    if: ${{ needs.gather-information.outputs.failed != 'true' }}
+    env:
+      PR_TITLE: ${{ github.event.pull_request.title }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          token: ${{ secrets.CONTANE_BOT_TOKEN }}
+
+      - name: Configure Git
+        run: |
+          git config user.name "contane-bot"
+          git config user.email "160241315+contane-bot@users.noreply.github.com"
+
+      - name: Install readme-generator-for-helm
+        if: ${{ needs.gather-information.outputs.values_changed == 'true' }}
+        run: npm install -g @bitnami/readme-generator-for-helm@2.6.1
+
+      - name: Update README.md
+        if: ${{ needs.gather-information.outputs.values_changed == 'true' }}
+        shell: bash
+        run: |
+          cd charts || exit 1
+          echo "Updating README.md for $CHANGED_CHART"
+          readme-generator --values "$CHANGED_CHART/values.yaml" --readme "$CHANGED_CHART/README.md"
+
+      - name: Update changelog.md
+        shell: bash
+        run: |
+          echo "Generating changelog for $CHANGED_CHART"
+          npx semantic-release --dry-run --no-ci --plugins @semantic-release/release-notes-generator > $CHANGED_CHART/release-notes.md
+          # Extract relevant part
+          sed -n '/### \[/{:a;n;/### \[/{p;q};p;ba}' $CHANGED_CHART/release-notes.md > $CHANGED_CHART/CHANGELOG.md
+          rm $CHANGED_CHART/release-notes.md
+
+      - name: Install changelog-generator
+        run: npm install -g github-changelog-generator
+
+      - name: Update ArtifactHub change annotations
+        shell: bash
+        run: |
+          # Parse PR title to create changelog entries
+          changes=""
+          if [[ $PR_TITLE == *"feat:"* ]]; then
+            changes="$changes"$'\n'  - kind: feature
+            changes="$changes"$'\n'    description: "${PR_TITLE#*: }"
```

Comment:
I've never seen this syntax before... What does the second $ do, and why are there no quotes surrounding `  - kind: feature` etc?

Reply:
Additionally, if there was a quote in the PR title, it would need to be escaped. We might have to use a tool for that, since any backslash in the PR title would also need to be escaped, etc. In general, this can often lead to invalid YAML. Not a security risk, just something that could lead to broken commits.